index.js

```diff
@@ -37,24 +38,68 @@ util.inherits(FilePort, Port);
 
 FilePort.prototype.init = function init() {
     Port.prototype.init.apply(this, arguments);
-
-    this.config = Object.keys(defaults).reduce(function(pv, cv) {
-        pv[cv] = this.config[cv] || defaults[cv].v;
-        return pv;
-    }.bind(this), this.config);
+    this.config.writeBaseDir = path.join(this.bus.config.workDir, 'ut-port-file', this.config.id);
 };
 
 FilePort.prototype.start = function start() {
     Port.prototype.start.apply(this, arguments);
-    this.stream = through2.obj(function(chk, enc, cb) {
-        this.push(chk);
-        cb();
+
+    return new Promise((resolve, reject) => fs.access(this.config.writeBaseDir, fs.R_OK | fs.W_OK, err => {
+        if (err) {
+            if (err.code === 'ENOENT') {
+                fs.makeTree(this.config.writeBaseDir, err => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve('Dir for journal log files has been created: ' + this.config.writeBaseDir);
+                    }
+                });
+            } else {
+                reject(err);
+            }
+        } else {
+            resolve('Dir for journal log files has been verified: ' + this.config.writeBaseDir);
+        }
+    })).then(result => {
+        this.stream = through2.obj(function(chk, enc, cb) {
+            this.push(chk);
+            cb();
+        });
+        this.streams = this.pipe(this.stream, {trace: 0, callbacks: {}});
+        this.pipeExec(this.exec.bind(this));
+        this.watch();
+        this.bindNotifier();
+        return result;
     });
-    this.streams = this.pipe(this.stream, {trace: 0, callbacks: {}});
+};
 
-    // start watching
-    this.watch();
-    this.bindNotifier();
+FilePort.prototype.exec = function exec({filename, data, encoding = 'utf8', append = true}) {
+    if (!filename || !data) {
+        return Promise.reject(errors.arguments());
+    }
+    if (path.isAbsolute(filename)) {
+        return Promise.reject(errors.absolutePath());
+    }
+    filename = path.join(this.config.writeBaseDir, filename);
+    if (path.resolve(filename.substr(0, this.config.writeBaseDir.length)) !== path.resolve(this.config.writeBaseDir)) {
```

Comment:
The idea is to avoid traversal outside the writeBaseDir. 
Unfortunately implementation is overcomplicated, while it could have been simply:
```js
filename = path.resolve(this.config.writeBaseDir, filename);
if (!filename.startsWith(this.config.writeBaseDir + path.sep)) {
```
@d0ntth1nc, please fix it.